### ISSUE TITLE
Wrong es6/es7 decorators indent 

### ISSUE THREAD

**Issue by stigmat4j** [pangloss/vim-javascript#326]
Before
![Image](http://croper.ru/images/201511294up2zSbDbBzHtg3z_3d73n_large.png?39973)

After
![Image](http://croper.ru/images/20151129sZhSQUIbNlmFEKOF_AaMzh_large.png?11971)

**Comment by WChargin** [comment]
Your images 404. Can you paste the text? The following indents properly for me:

```javascript
@pureRender
@apiRoute(
    '/path/to/thing',
)
class Thing {

    @magic
    method() {
    }

}
```

**Comment by robjens** [comment]
I'm having offbeat indents as well, or at least, I would suspect that e.g. a long destructuring statement a la

```js
 const {__, pathSatisfies, always, ifElse, props, identity, lensPath, set, flip,
    compose, isNil, isEmpty, either, complement, apply, equals} = R;
``` 
given a cursor position `= R;_` on enter would bring the cursor to the first level of the next line. Only it indents to the first position (4 spaces in my case).

```js
    compose, ... = R;
    ^
```

Arguably, the plugin, onmost occasions it seems to me, can ignore the presence of a `;` semicolon statement separator. This also goes for nested inside other statements, always off by one.

Weirdest I found something that ended up as follows (where one *would* actually expect it to respect the arguments indention):

```js
const initialize = ifElse (notNodeEnv, 
                                    compose (nodeEnvVal, set (lensEnv, 'development')),
nodeEnvVal);
```

where the correct position would be:

```js
const initialize = ifElse (notNodeEnv, 
                                    compose (nodeEnvVal, set (lensEnv, 'development')),
                                    nodeEnvVal);
```

as

```js
foo (bar,
       baz,
       bam);
```

Actually respects the indent but also falsely assumes that after the semicolon, we wish to maintain the indent. This is no such case of course, indent should be 0 ending `bam);` with a `[ENTER]`;

By the way, `set ai` is on. Which I guess it should be?

**Comment by robjens** [comment]
See where I referenced this, deleted the comment since I used to have a different version in my vundle and had different behavior. Apparently, it is the `select ... case` indent which screws up the multi-line variable (var, let, const) assignment. I noticed this because the other version I found didn't have this problem but originate from the same base. After adding in to that file, the (var|let|const) part that was missing, it works.

All that is left now, is to fix this bad boy:

```
const pprint = (o, {colors = true, level = 5} = {}) =>
    console.dir (o, {colors, level}); //<return>
/* _ <-- cursor end indent should be -1
_ <-- would be the right spot
```
